"Incompatible types" for asyncio.wait() + run_in_executor()

Sample program:
```
import asyncio
from typing import Any, Dict


async def message_reader(socket: Any) -> Any:
    return await socket.recv()


def do_work() -> Dict[str, Any]:
    return {}

async def foo(socket: Any) -> None:
    loop = asyncio.get_event_loop()
    work_task = loop.run_in_executor(None, do_work)
    recv_task = asyncio.ensure_future(message_reader(socket))
    pending = {work_task, recv_task}

    done, pending = await asyncio.wait(pending)
```

Error:
```
$ mypy --strict wait_mypy.py
wait_mypy.py:19: error: Incompatible types in assignment (expression has type "Set[Future[Any]]", variable has type "Set[Iterable[Any]]")
wait_mypy.py:19: error: Argument 1 to "wait" has incompatible type "Set[Iterable[Any]]"; expected "Iterable[Union[Generator[Any, None, <nothing>], Awaitable[<nothing>]]]"
wait_mypy.py:19: error: Argument 1 to "wait" has incompatible type "Set[Iterable[Any]]"; expected "Iterable[Union[Generator[Any, None, Any], Awaitable[Any]]]"
$ mypy --version
mypy 0.570
```

What's going on here is that one of the two tasks is inferred to be a `typing.AwaitableGenerator`, and the other is an `asyncio.Future`. In typeshed, both of these classes inherit from `Iterable` (possibly indirectly) and `Awaitable`, with the `Iterable` base first. Apparently, mypy infers the first common base class to be the common type if there are multiple common base classes. That's a reasonable thing to do, but in the case of futures, inferring `Awaitable` is probably much more commonly the right thing to do. 

To fix this, I'll move the Awaitable base classes first. I still get an error `samples/asyncwait.py:20: error: Incompatible types in assignment (expression has type "Set[Future[Any]]", variable has type "Set[Awaitable[Any]]")` on @neilconway's code, but that's a separate issue that's easily worked around by using a different variable name.

--- a/stdlib/3.4/asyncio/futures.pyi
+++ b/stdlib/3.4/asyncio/futures.pyi
@@ -26,7 +26,7 @@ class _TracebackLogger:
 if sys.version_info >= (3, 5):
     def isfuture(obj: object) -> bool: ...
 
-class Future(Iterable[_T], Awaitable[_T], Generic[_T]):
+class Future(Awaitable[_T], Iterable[_T]):
     _state = ...  # type: str
     _exception = ...  # type: BaseException
     _blocking = False

--- a/stdlib/3/typing.pyi
+++ b/stdlib/3/typing.pyi
@@ -173,7 +173,7 @@ class Coroutine(Awaitable[_V_co], Generic[_T_co, _T_contra, _V_co]):
 
 # NOTE: This type does not exist in typing.py or PEP 484.
 # The parameters corrrespond to Generator, but the 4th is the original type.
-class AwaitableGenerator(Generator[_T_co, _T_contra, _V_co], Awaitable[_V_co],
+class AwaitableGenerator(Awaitable[_V_co], Generator[_T_co, _T_contra, _V_co],
                          Generic[_T_co, _T_contra, _V_co, _S]):
     pass
 

